src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToHtml5LocalDateTimeTransformer.php

```diff
@@ -81,7 +98,15 @@ public function reverseTransform($dateTimeLocal)
             return;
         }
 
-        if (!preg_match('/^(\d{4})-(\d{2})-(\d{2})[T ]\d{2}:\d{2}(?::\d{2})?$/', $dateTimeLocal, $matches)) {
+        // to maintain backwards compatibility we do not strictly validate the submitted date
+        // see https://github.com/symfony/symfony/issues/28699
+        if (!$this->strict) {
+            $regex = '/^(\d{4})-(\d{2})-(\d{2})[T ]\d{2}:\d{2}(?::\d{2})?/';
```

Comment:
to maintain BC, shouldn't it use the old transformer instead (which did not have that regex either, so you are only partially fixing the BC break)